### PR TITLE
[bugfix] Clear realtime inverted index bitmaps on close

### PR DESCRIPTION
**Pull request by xiangfu0**
## Summary
- make `RealtimeInvertedIndex.close()` actively clear and trim in-memory bitmap state
- add a regression test that verifies `close()` empties the bitmaps and returns empty results afterward

## Root cause
`MutableSegmentImpl.destroy()` already walks mutable indexes and calls `close()`, but `RealtimeInvertedIndex.close()` was a no-op. After a consuming segment was persisted and replaced, any lingering wrapper or data-source reference could still retain the in-memory posting lists because the index never dropped its bitmap references.

## Reproduction
1. Create a consuming segment with a dictionary-encoded column and realtime inverted index enabled.
2. Index rows so the mutable inverted index allocates posting lists.
3. Persist and replace the segment, then let the mutable segment destroy path run.
4. Inspect the heap and follow a stale `indexContainer` or `DataSource` wrapper: the `RealtimeInvertedIndex` still retains its `_bitmaps` list because `close()` does not release anything.

## Impact
This makes realtime inverted-index cleanup deterministic once the mutable segment destroy path runs, reducing retained heap after realtime segment replacement.

## Validation
- `./mvnw spotless:apply -pl pinot-segment-local`
- `./mvnw -pl pinot-segment-local -Dtest=RealtimeInvertedIndexReaderTest test`
- `./mvnw checkstyle:check -pl pinot-segment-local`
- `./mvnw license:format -pl pinot-segment-local`
- `./mvnw license:check -pl pinot-segment-local`

### CODE REVIEW

File: pinot-segment-local/src/main/java/org/apache/pinot/segment/local/realtime/impl/invertedindex/RealtimeInvertedIndex.java [hunk 1]
```diff
@@ -82,5 +82,11 @@ public MutableRoaringBitmap getDocIds(int dictId) {
 
   @Override
   public void close() {
+    try {
+      _writeLock.lock();
+      _bitmaps.clear();
+    } finally {
```
[the remark this answers is not in the document]
Comment (noob-se7en) [reply pinot-segment-local/src/main/java/org/apache/pinot/segment/local/realtime/impl/invertedindex/RealtimeInvertedIndex.java:88]: +1

**Review by noob-se7en — APPROVED** [review]

**Review by Jackie-Jiang — COMMENTED** [review]
This is not really the contract of `close()`. The real issue is the dangling reference.
Do you see where the dangling reference is from? We should find that as the root cause.
If you want to ensure all the references to the index are removed when destroying the `MutableSegmentImpl`, you can clear the `_indexContainerMap` during the destroy.

**Comment by xiangfu0** [comment]
> This is not really the contract of `close()`. The real issue is the dangling reference. Do you see where the dangling reference is from? We should find that as the root cause. If you want to ensure all the references to the index are removed when destroying the `MutableSegmentImpl`, you can clear the `_indexContainerMap` during the destroy.

correct, this is more on the defensive side, even if the caller side hold the MutableSegment or Index reference, the leakage for memory is minimal.
